File: forge/src/main/java/com/rpmtw/rpmtw_platform_mod/forge/RPMTWPlatformModForge.java
package com.rpmtw.rpmtw_platform_mod.forge;

import com.rpmtw.rpmtw_platform_mod.RPMTWPlatformMod;
import net.minecraftforge.fml.common.Mod;

@Mod(RPMTWPlatformMod.MOD_ID)
public class RPMTWPlatformModForge {
    public RPMTWPlatformModForge() {
        ForgeKotlinClient forgeClient = new ForgeKotlinClient();
        forgeClient.onInitialize();
    }

}

File: forge/src/main/kotlin/com/rpmtw/rpmtw_platform_mod/forge/RPMTWPlatformModForge.kt
package com.rpmtw.rpmtw_platform_mod.forge

import com.rpmtw.rpmtw_platform_mod.RPMTWPlatformMod
import com.rpmtw.rpmtw_platform_mod.RPMTWPlatformMod.init
import dev.architectury.platform.forge.EventBuses
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.event.lifecycle.FMLCommonSetupEvent
import thedarkcolour.kotlinforforge.forge.MOD_BUS
import thedarkcolour.kotlinforforge.forge.MOD_CONTEXT

@Mod(RPMTWPlatformMod.MOD_ID)
class RPMTWPlatformModForge {
    init {
        EventBuses.registerModEventBus(RPMTWPlatformMod.MOD_ID, MOD_CONTEXT.getKEventBus())
        MOD_BUS.addListener(this::init)
    }

    private fun init(event: FMLCommonSetupEvent) {
        init()
    }

}